lib/src/spec/message.dart

```diff
@@ -49,38 +50,71 @@ class MessageData<T> {
   }
 }
 
+/// Delta extension configuration for [MessageExtras]
+class DeltaExtras {
+  /// the id of the message the delta was generated from
+  final String from;
+
+  /// the delta format. Only "vcdiff" is supported currently
+  final String format;
+
+  /// initializes with from and format
+  const DeltaExtras({this.from, this.format});
+
+  /// create instance from a map
+  static DeltaExtras _fromMap(Map value) {
+    if (value == null) return null;
+    return DeltaExtras(
+      from: value['from'] as String,
+      format: value['format'] as String,
+    );
+  }
+
+  @override
+  bool operator ==(Object other) =>
+      other is DeltaExtras && other.from == from && other.format == format;
+
+  @override
+  int get hashCode => '$from:$format'.hashCode;
+}
+
 /// Handles supported message extras types, their encoding and decoding
-class MessageExtras<T> {
-  final T _extras;
+class MessageExtras {
+  /// json-encodable map of extras
+  final Map<String, dynamic> extras;
+
+  /// configuration for delta compression extension
+  DeltaExtras _delta;
 
-  /// Only Map and List types are supported
-  MessageExtras(this._extras) : assert(T == Map);
+  /// delta configuration received from channel message
+  DeltaExtras get delta => _delta;
```

Comment:
This can be `null` - are we not able to annotate that on the interface?

Reply:
We can do that with `nullsafety` migration

Reply:
see: https://github.com/ably/ably-flutter/issues/82